compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -3758,13 +3758,13 @@ fn hint_missing_borrow<'tcx>(
     err: &mut Diagnostic,
 ) {
     let found_args = match found.kind() {
-        ty::FnPtr(f) => f.inputs().skip_binder().iter(),
+        ty::FnPtr(f) => infcx.replace_bound_vars_with_placeholders(*f).inputs().iter(),
```

Comment:
This replaces bounds vars in the entire `FnSig` and then gets only the inputs. Could also do

```rust
f.inputs().iter()
```

and do the `replace_bound_vars_with_placeholders` inside `get_deref_type_and_refs`...

or potentially even keep the binder around all the way into `can_eq`.